torchdynamo/_guards.cpp

```diff
@@ -151,8 +191,38 @@ PyObject *TensorGuards_check(TensorGuards *self, PyObject *args) {
   Py_RETURN_TRUE;
 }
 
+PyObject *TensorGuards_check_verbose(TensorGuards *self, PyObject *args) {
+  if (!PyTuple_CheckExact(args)) {
+    PyErr_SetString(PyExc_TypeError, "expected tuple()");
+    return NULL;
+  }
+  auto &checks = *self->checks;
+  ssize_t len = PyTuple_GET_SIZE(args);
+
+  if (static_cast<ssize_t>(checks.size()) != len) {
+    PyErr_SetString(PyExc_TypeError, "wrong length");
+    return NULL;
+  }
+
+  LocalState state;
+
+  for (ssize_t i = 0; i < len; ++i) {
```

Comment:
it seems more conceivable that we would introduce some branches in `TensorGuards_check` to avoid code duplication, and keep the 'check_verbose' part separate?

Reply:
I'd be fine with this as long we measure the overheads to make sure it doesn't hurt things